src/worldstate/CMakeLists.txt

```diff
@@ -28,30 +28,70 @@ add_message_files(
 
 ## Generate added messages and services with any dependencies listed here
 generate_messages(
-  DEPENDENCIES
-  std_msgs
+  DEPENDENCIES std_msgs
 )
 
 ###################################
 ## catkin specific configuration ##
 ###################################
 
 catkin_package(
+   INCLUDE_DIRS include
    CATKIN_DEPENDS message_runtime
 )
 
+###########
+## Build ##
+###########
+
+include_directories(
+        ${catkin_INCLUDE_DIRS}
+        ./include
+        ./routine/include
+        ./routine/Gate/include
+        ../constants
+)
+
+add_executable(world_state_node
+    src/WorldStateNode.cpp
+    src/world_state_node.cpp
+    include/WorldStateNode.h
+    routine/include/State.h
+    routine/src/State.cpp
+    routine/Gate/include/locatingGate.h
+    routine/Gate/src/locatingGate.cpp
+    routine/Gate/include/alignWithGate.h
+    routine/Gate/src/alignWithGate.cpp
+    routine/Gate/include/passGate.h
+    routine/Gate/src/passGate.cpp
```

Comment:
Our convention (and a pretty standard C++ one), is to have the `src` and `include` folders both at top level, ie

`routine/Gate/include/locatingGate.h` -> `include/routine/Gate/locatingGate.h`
`routine/Gate/src/locatingGate.cpp` -> `src/routine/Gate/locatingGate.cpp`

Reply:
I'd also suggest renaming `routine` to `routines`, as I think it's a bit more clear. (We're defining multiple routines, the `routine` folder doesn't just contain a single one).

Reply:
Changed!